src/entities/QuarterBudget/model.ts

```diff
@@ -195,4 +209,25 @@ export default class QuarterBudgetModel extends Model<QuarterBudgetAttributes> {
       categories: categoriesResults,
     }
   }
+
+  static parseBudgets(results: BudgetQueryResult[]): Budget[] {
+    const budgetsByQuarter = new Map<string, BudgetQueryResult[]>()
+
+    results.forEach((result) => {
+      const { id } = result
+      const group = budgetsByQuarter.get(id)
+
+      if (group) {
+        group.push(result)
+      } else {
+        budgetsByQuarter.set(id, [result])
+      }
+    })
+
+    const budgets: Budget[] = []
+
+    budgetsByQuarter.forEach((resultsForQuarter) => budgets.push(this.parseBudget(resultsForQuarter)))
```

Comment:
Can't this be done with a map function?
`const budgets = budgetsByQuarter.map(resultsForQuarter => this.parseBudget(resultsForQuarter)`?

Reply:
The `Map` class does not contain the map() function so it would have to be transformed first to an Array to do that

It would be something like: `Array.from(budgetsByQuarter.values()).map(...)`